site/content/en/docs/Advanced/controlling-disruption.md

```diff
@@ -0,0 +1,113 @@
+
+Note that to maintain backwards compatibility with Agones prior to the introduction of the `SafeToEvict` feature gate, if your game server previously configured the `cluster-autoscaler.kubernetes.io/safe-to-evict: true` annotation, we assume `eviction.safe: Always` is intended.
+
+### `safe: UpgradesOnly`
+
+If your game server has a session length between ten minutes and one hour, you should set:
+
+```
+eviction:
+  safe: UpgradesOnly
+```
+
+This will ensure the game server `Pod` can be evicted by node upgrades, but Cluster Autoscaler will not evict it.
+
+{{< alert title="GKE Autopilot" color="info" >}}
+GKE Autopilot supports only `Never` and `Always`, not `UpgradeOnly`.
+{{< /alert >}}
+
+## What's special about ten minutes and one hour?
+
+* **Ten minutes:** Cluster Autoscaler respects [ten minutes of graceful termination](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#does-ca-respect-gracefultermination-in-scale-down) on scale-down. On some cloud products, you can configure `--max-graceful-termination-sec` to change this, but it is not advised: Cluster Autoscaler is currently only capable of scaling down one node at a time, and larger graceful termination windows slow this down farther (see [autoscaler#5079](https://github.com/kubernetes/autoscaler/issues/5079)). If the ten minute limit does not apply to you, generally you should choose between `safe: Always` (for sessions less than an hour), or see [below](#considerations-for-long-sessions).
+
+* **One hour:** On many cloud products, `PodDisruptionBudget` can only block node upgrade evictions for a certain period of time - on GKE this is 1h. After that, the PDB is ignored, or the node upgrade fails with an error. Controlling `Pod` disruption for longer than one hour requires cluster configuration changes outside of Agones - see [below](#considerations-for-long-sessions).
+
+## Considerations for long sessions
+
+Outside of Cluster Autoscaler, the main source of disruption for long sessions is node upgrade. On some cloud products, such as GKE Standard, node upgrades are entirely within your control. On others, such as GKE Autopilot, node upgrade is automatic. Typical node upgrades use an eviction based, rolling recreate strategy, and may not honor `PodDisruptionBudget` for longer than an hour. Here we document strategies you can use for your cloud product to support long sessions.
+
+### On GKE
+
+On GKE, there are currently two possible approaches to manage disruption for session lengths longer than an hour:
+
+* (GKE Standard/Autopilot) [Blue/green deployment](https://martinfowler.com/bliki/BlueGreenDeployment.html) at the cluster level: If you are using an automated deployment process, you can:
+  * create a new, `blue` cluster within a release channel e.g. every week,
```

Comment:
nit-picky: usually the `blue` refers to the new resource and `green` refers to the old one. The links you add here from "martinfowler.com" and "cloud.google.com" follow this way.

Reply:
Yeah, I totally reversed those.